### PR TITLE
doc/release-notes: add jewel->kraken notes

### PR DESCRIPTION
**Pull request by liewegas**
Forgot to include this!

Also add a bit more warning around the 'crush location' option change.

Signed-off-by: Sage Weil <sage@redhat.com>

### CODE REVIEW

File: doc/release-notes.rst [hunk 1]
```diff
@@ -610,6 +610,130 @@ Upgrading from pre-Jewel releases (like Hammer)
 You *must* first upgrade to Jewel (10.2.z) before attempting an
 upgrade to Luminous.
 
+Upgrade compatibility notes, Jewel to Kraken
+--------------------------------------------
+
+These changes occurred between the Jewel and Kraken releases and will affect
+upgrades from Jewel to Luminous.
+
+* The `osd crush location` config option is no longer supported.  Please
```
Comment (theanalyst) [inline doc/release-notes.rst]: is this applicable post 12.2.1 or for all of 12.2.x series?

File: doc/release-notes.rst [hunk 2]
```diff
@@ -610,6 +610,130 @@ Upgrading from pre-Jewel releases (like Hammer)
...
+  update your ceph.conf to use the `crush location` option instead.  Be sure
+  to update your config file to avoid any movement of OSDs from your customized
+  location back to the default one.
```
Comment (wido) [inline doc/release-notes.rst]: Looks good to me!

File: doc/release-notes.rst (hunk 1 above)
Comment (alfredodeza) [inline doc/release-notes.rst]: this needs double backticks, not single

File: doc/release-notes.rst [hunk 3]
```diff
@@ -610,6 +610,130 @@ Upgrading from pre-Jewel releases (like Hammer)
...
 upgrade to Luminous.
 
+Upgrade compatibility notes, Jewel to Kraken
+--------------------------------------------
+
+These changes occurred between the Jewel and Kraken releases and will affect
+upgrades from Jewel to Luminous.
+
+* The `osd crush location` config option is no longer supported.  Please
+  update your ceph.conf to use the `crush location` option instead.  Be sure
```
Comment (alfredodeza) [inline doc/release-notes.rst]: double backticks

File: doc/release-notes.rst [hunk 4]
```diff
@@ -610,6 +610,130 @@ Upgrading from pre-Jewel releases (like Hammer)
...
+  Please switch to using just 'jerasure' or 'shec'.
+
+* The librados omap get_keys and get_vals operations include a start key and a
+  limit on the number of keys to return.  The OSD now imposes a configurable
+  limit on the number of keys and number of total bytes it will respond with,
+  which means that a librados user might get fewer keys than they asked for.
+  This is necessary to prevent careless users from requesting an unreasonable
+  amount of data from the cluster in a single operation.  The new limits are
+  configured with `osd_max_omap_entries_per_request`, defaulting to 131,072, and
+  'osd_max_omap_bytes_per_request', defaulting to 4MB.
```
Comment (alfredodeza) [inline doc/release-notes.rst]: both these `osd_` values look like they need double backticks too

File: doc/release-notes.rst [hunk 5]
```diff
@@ -4114,8 +4238,8 @@ Upgrade notes
   change it to a negative value (e.g., -1) to preserve the current
   behavior.
 
-* The `osd crush location` config option is no longer supported.  Please
-  update your ceph.conf to use the `crush location` option instead.
+* The ``osd crush location`` config option is no longer supported.  Please
+  update your ceph.conf to use the ``crush location`osd_` option instead.
```
Comment (theanalyst) [inline doc/release-notes.rst]: osd_ is creating a docs build error

**Comment by liewegas** [comment]
@alfredodeza builds now!

File: doc/release-notes.rst [hunk 6]
```diff
@@ -610,6 +610,130 @@ Upgrading from pre-Jewel releases (like Hammer)
...
+  new OSD.  Now, a value of 0 means it should have a weight of 0, and
+  a negative value (the new default) means we should automatically
+  weight the OSD based on its size.  If your configuration file
+  explicitly specifies a value of 0 for this option you will need to
+  change it to a negative value (e.g., -1) to preserve the current
+  behavior.
+
+* The static libraries are no longer included by the debian
+  development packages (lib*-dev) as it is not required per debian
+  packaging policy.  The shared (.so) versions are packaged as before.
```
Comment (smithfarm) [inline doc/release-notes.rst]: The "static libraries are no longer included by the debian. . ." release note can be dropped.

File: doc/release-notes.rst [hunk 7]
```diff
@@ -610,6 +610,130 @@ Upgrading from pre-Jewel releases (like Hammer)
...
+
+* The libtool pseudo-libraries (.la files) are no longer included by
+  the debian development packages (lib*-dev) as they are not required
+  per https://wiki.debian.org/ReleaseGoals/LAFileRemoval and
+  https://www.debian.org/doc/manuals/maint-guide/advanced.en.html.
```
Comment (smithfarm) [inline doc/release-notes.rst]: The "libtool pseudo-libraries are no longer included by the debian . . ." RN can be dropped.

**Pull request merged**